components/turret.py

```diff
@@ -91,6 +91,17 @@ def _slew_to_count(self, count: int) -> None:
             self.incrementing = False
         self.current_state = self.SLEWING
 
+    def scan(self, heading):
+        """
+        Slew the turret back and forth looking for a target.
+        """
+        # If we haven't hit an index yet, we just have to scan
+        # about the current position.
+        # Otherwise scan about the heading we've been given.
+        # The target must be in front of us, so scan up to
```

Comment:
Why must the target be in front of us? If we are going for a ball on the field, we might very well be heading away from the target. Scanning 90 degrees on either side still seems a reasonable thing to do, though, so I'd just cut the bit about the target being in front.

Reply:
Downfield is a better term
It has to be in front of the robot from the driver station perspective, not from the x axis of the robot.